File: Goodreads Flutter/lib/domain/Book.dart
class Book {
  String id;
  String title;
  String author;
  String description;
  int rating;

  Book(this.title, this.author, this.description, this.rating);
  Book.withId(this.id, this.title, this.author, this.description, this.rating);

  Map<String, dynamic>  toMap() {
    var map = Map<String, dynamic>();

    if (id != null) {
      map['id'] = id;
    }
    map['title'] = title;
    map['description'] = description;
    map['author'] = author;
    map['rating'] = rating.toString();

    return map;
  }

  Book.fromMap(Map<String, dynamic> map) {
    this.id = map['id'].toString();
    this.title = map['title'];
    this.description = map['description'];
    this.author = map['author'];
    this.rating = int.parse(map['rating'].toString());
  }
}

File: GoodreadsL3/app/src/main/java/com/gecopuf/george/goodreads/model/Book.kt
package com.gecopuf.george.goodreads.model


class Book {
    var id: String? = null
    var title: String? = null
    var author: String? = null
    var description: String? = null
    var rating: Int? = null

    constructor() {}

    constructor(id: String, title: String, author: String, description: String, rating: Int) {
        this.id = id
        this.title = title
        this.author = author
        this.description = description
        this.rating = rating
    }

    constructor(title: String, author: String, description: String, rating: Int) {
        this.title = title
        this.author = author
        this.description = description
        this.rating = rating
    }

    fun toMap(): Map<String, Any> {
        val result = HashMap<String, Any>()
        result["title"] = title!!
        result["author"] = author!!
        result["description"] = description!!
        result["rating"] = rating!!

        return result
    }
}